File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderBooking.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.annotations.ApiModel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.io.Serializable;

@ApiModel(description = "Offender Booking Summary")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder(toBuilder = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
public class OffenderBooking implements Serializable {

    private Long bookingId;
    private String bookingNo;
    private String offenderNo;
    private String imprisonmentStatus;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/OffenderBooking.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude
import org.springframework.data.redis.core.RedisHash
import java.io.Serializable

@RedisHash("OffenderBooking")
@JsonInclude(JsonInclude.Include.NON_NULL)
data class OffenderBooking(
  val bookingId: Long? = null,
  val bookingNo: String? = null,
  val offenderNo: String? = null,
  val imprisonmentStatus: String? = null,
) : Serializable
